src/lib.rs

```diff
@@ -377,12 +377,21 @@ where
 pub(crate) fn domain(
     request: &tungstenite::handshake::client::Request,
 ) -> Result<String, tungstenite::Error> {
-    match request.uri().host() {
-        Some(d) => Ok(d.to_string()),
-        None => Err(tungstenite::Error::Url(
-            tungstenite::error::UrlError::NoHostName,
-        )),
-    }
+    request
+        .uri()
+        .host()
+        .map(|host| {
+            // If host is an IPv6 address, it might be surrounded by brackets. These brackets are
+            // *not* part of a valid IP, so they must be stripped out.
+            let host = if host.starts_with('[') {
+                &host[1..host.len() - 1]
```

Comment:
This will panic if an invalid URI like `wss://[` is given. Can you add some more checks? Maybe work with `str::strip_prefix()` / `str::strip_suffix()` here

Reply:
I don't think we need those checks, since `host` is being extracted from a valid `http::Uri`. An invalid URI would be caught when trying to construct the `Request`.

Reply:
In that case please add a comment about that and a test that also checks for this very condition :)

Reply:
Sure thing, can you clarify a bit what you'd like that test to look like?

Reply:
Check if `ws://[` or `ws://[blabla/bla` or `ws://[::1/bla` are correctly rejected without a panic, either by the URI parser or by this function

Reply:
Or other variations if you can think of some :)

I wonder if the bracket handling for IPv6 shouldn't be part of the URI parser though...

Reply:
Updated.

> I wonder if the bracket handling for IPv6 shouldn't be part of the URI parser though...

It was an intentional decision: https://github.com/hyperium/http/issues/291. The nuance here is that the brackets are part of a valid *URI*, but not a valid *IP*.

Reply:
Makes sense, thanks!